exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -93,7 +95,6 @@ func (b *Bitfinex) wsReadData(ws stream.Connection) {
 
 // WsDataHandler handles data from wsReadData
 func (b *Bitfinex) WsDataHandler() {
-	b.Websocket.Wg.Add(1)
 	defer b.Websocket.Wg.Done()
 	for {
 		select {
```

Comment:
This will also need a `comms` drainer on shutdown. Also please place `ShutdownC` as the first `case`